src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -0,0 +1,354 @@
+                csvValues[transformFieldIndex] = transformedValue;
+
+                // Write line to table (plus new line char).
+                writer.write(csvValues);
+
+                // Count number of CSV rows changed.
+                if (!originalValue.equals(transformedValue)) {
+                    modifiedRowCount++;
+                }
+            } // End of iteration over each row.
+            csvReader.close();
+            writer.flush();
+
+            // Copy csv input stream into the zip file, replacing the existing file.
+            try (
+                // Modify target zip file that we just read.
+                FileSystem targetZipFs = FileSystems.newFileSystem(tempZipPath, null);
+                // Stream for file copy operation.
+                InputStream inputStream =  new ByteArrayInputStream(stringWriter.toString().getBytes(StandardCharsets.UTF_8))
+            ) {
+                Path targetTxtFilePath = getTablePathInZip(tableName, targetZipFs);
+                Files.copy(inputStream, targetTxtFilePath, StandardCopyOption.REPLACE_EXISTING);
+                zipTarget.feedTransformResult.tableTransformResults.add(
+                    new TableTransformResult(tableName, 0, modifiedRowCount, 0)
+                );
+            }
+
+            // Replace original zip file with temporary working zip file.
+            // (This should also trigger a system IO update event, so subsequent IO calls pick up the correct file.
+            Files.move(tempZipPath, originalZipPath, StandardCopyOption.REPLACE_EXISTING);
+            LOG.info("Field normalization transformation successful ({} row(s) changed).", modifiedRowCount);
+        } catch (Exception e) {
+            status.fail("Unknown error encountered while transforming zip file", e);
+        }
+    }
+
+    /**
+     * Converts the provided string to Title Case,
+     * accommodating for separator characters that may be immediately precede
+     * (without spaces) names that we need to capitalize.
+     */
+    public String convertToTitleCase(String inputString) {
+        // Return at least a blank string.
+        if (StringUtils.isBlank(inputString)) {
+            return "";
+        }
+
+        String result = WordUtils.capitalizeFully(inputString, SEPARATORS);
+
+        // Exceptions (e.g. acronyms) should remain capitalized.
+        for (Substitution substitution : getCapitalizeSubstitutions()) {
+            result = substitution.replace(result);
+        }
+
+        return result;
+    }
+
+    /**
+     * Performs substitutions defined for this transform, and returns the result.
+     */
+    public String performSubstitutions(String inputString) {
+        String result = inputString;
+        for (Substitution substitution : getStringSubstitutions()) {
+            result = substitution.replace(result);
+        }
+        return result;
+    }
+
+    public enum NormalizeOperation {
+        SENTENCE_CASE, TITLE_CASE
+    }
```

Comment:
This doesn't appear to be used anywhere. Is it needed?

Reply:
I think I included this in the original stubbed out branch. I think it might be good to default the normalization operation to TITLE_CASE and keep the enum in place for eventually extending it to other types (if needed)? Just trying to keep extensibility in mind, but I don't feel too strongly about this if others dissent.